app/services/commits_runner.rb

```diff
@@ -1,11 +1,43 @@
-class CommitsRunner
-  def self.run(commits, pattern = '')
-    commits.select { |commit| valid?(commit) }
-           .each { |commit| create_and_run(commit, pattern) }
-           .count
+module CommitsRunner
+  def self.run(trigger_source, commits, repo, pattern = '')
+    formatted_commits = if trigger_source == :webhook
+                          format_webhook(commits, repo)
+                        elsif trigger_source == :api
+                          format_api(commits, repo)
+                        end
+    
+    formatted_commits.select { |commit| valid?(commit) }
+                     .each { |commit| create_and_run(commit, pattern) }
+                     .count
   end
 
   private
+  
+  def self.format_api(commits, repo)
+    commits.map do |commit|
+      {
+        sha: commit['sha'],
+        message: commit['commit']['message'],
+        repo: repo,
+        url: commit['html_url'],
+        created_at: commit['commit']['author']['date'],
+        author_name: commit['commit']['author']['name']
+      }
+    end
+  end
+
+  def self.format_webhook(commits, repo)
+    commits.map do |commit|
+      {
+        sha: commit['id'],
+        message: commit['message'],
+        repo: repo,
+        url: commit['url'],
+        created_at: commit['timestamp'],
+        author_name: commit['author']['name'] 
+      }
+    end
```

Comment:
These formatting methods are needed since json fetched from api and webhook is formatted differently.